File: Credit (Python)/credit.py
import cs50

while True:
    cardnum = cs50.get_int("Number: ")
    if cardnum > 1:
        numstring = str(cardnum)
        ndigits = int(len(numstring))
        firstnumber = int(str(cardnum)[:1])
        firsttwonumbers = int(str(cardnum)[:2])
        checksum = 0
        for i in range(ndigits, -1, -2):
            thedouble = int(int((cardnum % 100) / 10) * 2)
            checksum += int((thedouble % 100) / 10) + int(thedouble % 10)
            checksum += int(cardnum % 10)
            cardnum /= 100
        if ndigits == 15 and (firsttwonumbers == 34 or firsttwonumbers == 37) and checksum % 10 == 0:
            print("AMEX")
        elif ndigits == 16 and (51 <= firsttwonumbers and firsttwonumbers <= 55) and checksum % 10 == 0:
            print("MASTERCARD")
        elif (ndigits == 16 or ndigits == 13) and firstnumber == 4 and checksum % 10 == 0:
            print("VISA")
        else:
            print("INVALID")
        break

File: Credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    long cardnum;
    long cardnum1;
    int checksum;
    double ndigits;
    int firstnumber;
    int firsttwonumbers;
    int thedouble;
    do
    {
        cardnum = get_long("Number: ");
        checksum = 0;
        ndigits = 0;
        
    }
    while (cardnum < 1);
    cardnum1 = cardnum;
    do
    {
        ndigits++;
        cardnum1 /= 10;
    }
    while (cardnum1 > 0);
    firstnumber = cardnum / (1 * pow(10, ndigits - 1));
    firsttwonumbers = cardnum / (1 * pow(10, ndigits - 2));
    for (int i = 0; i <= cardnum; i++)
    {
        thedouble = (cardnum % 100) / 10 * 2;
        checksum += (thedouble % 100) / 10 + thedouble % 10;
        checksum += (cardnum % 10);
        cardnum /= 100;
    }
    checksum += cardnum;
    if (ndigits == 15 && (firsttwonumbers == 34 || firsttwonumbers == 37) && checksum % 10 == 0)
    {
        printf("AMEX\n");
    }
    else if (ndigits == 16 && (51 <= firsttwonumbers && firsttwonumbers <= 55) && checksum % 10 == 0)
    {
        printf("MASTERCARD\n");
    }
    else if ((ndigits == 16 || ndigits == 13) && firstnumber == 4 && checksum % 10 == 0)
    {
        printf("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }
}